diesel/src/pg/expression/expression_methods.rs

```diff
@@ -683,3 +683,592 @@ where
+     * # }
+     * #
+     * # fn run_test() -> QueryResult<()> {
+     * #     use self::hosts::dsl::*;
+     * #     use ipnetwork::IpNetwork;
+     * #     use std::str::FromStr;
+     * #     let conn = establish_connection();
+     * #     conn.execute("DROP TABLE IF EXISTS hosts").unwrap();
+     * #     conn.execute("CREATE TABLE hosts (id SERIAL PRIMARY KEY, address INET NOT NULL)").unwrap();
+     * diesel::insert_into(hosts)
+     *     .values(vec![address.eq(IpNetwork::from_str("10.0.2.3").unwrap())])
+     *     .execute(&conn)?;
+     *
+     * let addr = hosts
+     *     .select(address.or(IpNetwork::from_str("0.0.0.255").unwrap()))
+     *     .first::<IpNetwork>(&conn)?;
+     * assert_eq!(addr, IpNetwork::from_str("10.0.2.255").unwrap());
+     * #     Ok(())
+     * # }
+     * ```
+     */
+    fn or<T>(self, other: T) -> dsl::OrNet<Self, T>
+    where
+        T: AsExpression<Inet>,
+    {
+        Grouped(OrNet::new(self, other.as_expression()))
+    }
+}
+
+impl<T> PgNetExpressionMethods for T
+where
+    T: Expression,
+    T::SqlType: InetOrCidr,
+{
+}
+
+/// PostgreSQL specific methods present between CIDR/INET expressions and Bigint expressions
+pub trait PgNetAddExpressionMethods: Expression + Sized {
+    /**
+     * Creates a PostgreSQL `+` expression.
+     *
+     * This operator adds an offset to an address
+     *
+     * # Example
+     *
+     * ```rust
+     * # #[macro_use] extern crate diesel;
+     * # extern crate ipnetwork;
+     * # include!("../../doctest_setup.rs");
+     * #
+     * # table! {
+     * #     hosts {
+     * #         id -> Integer,
+     * #         address -> Inet,
+     * #     }
+     * # }
+     * #
+     * # fn main() {
+     * #     run_test().unwrap();
+     * # }
+     * #
+     * # fn run_test() -> QueryResult<()> {
+     * #     use self::hosts::dsl::*;
+     * #     use ipnetwork::IpNetwork;
+     * #     use std::str::FromStr;
+     * #     let conn = establish_connection();
+     * #     conn.execute("DROP TABLE IF EXISTS hosts").unwrap();
+     * #     conn.execute("CREATE TABLE hosts (id SERIAL PRIMARY KEY, address INET NOT NULL)").unwrap();
+     * diesel::insert_into(hosts)
+     *     .values(vec![address.eq(IpNetwork::from_str("10.0.2.3").unwrap())])
+     *     .execute(&conn)?;
+     *
+     * let addr = hosts
+     *     .select(address.add(10))
+     *     .first::<IpNetwork>(&conn)?;
+     * assert_eq!(addr, IpNetwork::from_str("10.0.2.13").unwrap());
+     * #     Ok(())
+     * # }
+     * ```
+     */
+    fn add<T>(self, other: T) -> dsl::AddNet<Self, T>
+    where
+        T: AsExpression<Bigint>,
+    {
+        Grouped(AddNet::new(self, other.as_expression()))
+    }
+}
+
+impl<T> PgNetAddExpressionMethods for T
+where
+    T: Expression,
+    T::SqlType: InetOrCidr,
+{
+}
+
+/// PostgreSQL specific methods present between CIDR/INET expressions and Bigint expression
+pub trait PgNetSubExpressionMethods: Expression + Sized {
```

Comment:
The comment from add also applies here.

Reply:
Same as above.

Reply:
I tried the `Sub` out, and there is an issued regarding the fact that the `-` operator works for two Rhs types:
- `Inet - Bigint -> Inet` (`.sub()`)
- `Inet - Inet -> Biging` (`.diff()`)

As per the doc of Diesel's ops traits, it is not possible to have two Rhs types (and indeed I could not make it work), so one of them will need to be kept the current way (either `.sub()` or `.diff()`).

Personally, I'd rather keep `.diff()` and transition `.sub()` to the `Sub` trait, as to keep the consistency with `Add` which has a `Bigint` Rhs. Any preference?

Reply:
Done in d486715c2. I did it for the previous `sub()` and kept `.diff()`

Reply:
:+1: